opentimestamps/core/secp256k1.py

```diff
@@ -0,0 +1,145 @@
+# Copyright (C) 2017 The OpenTimestamps developers
+#
+# This file is part of python-opentimestamps.
+#
+# It is subject to the license terms in the LICENSE file found in the top-level
+# directory of this distribution.
+#
+# No part of python-opentimestamps including this file, may be copied,
+# modified, propagated, or distributed except according to the terms contained
+# in the LICENSE file.
+
+import hashlib
+
+from opentimestamps.core.op import UnaryOp, MsgValueError
+
+@UnaryOp._register_op
+class OpSecp256k1Commitment(UnaryOp):
+    """Map (P || commit) -> [P + sha256(P||commit)G]_x for a given secp256k1 point P
+
+    This is a unary op rather than a binary op to allow timestamps to also
+    timestamp the point itself; in the event of an ECC break this might be
+    relevant. Such a break would not affect the integrity of the commitment,
+    but knowledge of the underlying key may be interesting in its own right.
+    """
+    TAG = b'\x09'
+    TAG_NAME = 'secp256k1commitment'
+
+    def _do_op_call(self, msg):
+        if len(msg) < 33:
+            raise MsgValueError("Missing secp256k1 point")
+
+        pt = Point.decode(msg[0:33])
+
+        hasher = hashlib.sha256()
+        hasher.update(pt.encode())
```

Comment:
@apoelstra Question: could `msg[0:33]` ever be different than the output from `pt.encode()`? And what prevents two encoded points from mapping to the same decoded then encoded point?

Reply:
@petertodd 

1. Sure. But anything that doesn't `throw` will be identical to the output of `pt.encode()` for some point.
2. The encoding includes the full x coordinate and the sign of the y coordinate of a point. If two points have the same x-coord and y-sign then they are the same point.

Reply:
Ah, I see. Yep, they will always be identical so if you're more comfortable using `msg[0:33]` I can do that instead.

Reply:
Sounds like a good idea.

Also, if we think this should be mathematically impossible, maybe adding an assertion that tests this isn't a bad idea. Worst case is a DoS attack in a situation where something is quite wrong in our understanding.